### PR TITLE
Bug/dr 12284 fixes reuploading same file after removal

### PR DESCRIPTION
**Pull request by aholzner**

### CODE REVIEW

File: src/components/VueBucketLoader.vue [hunk 1]
```diff
@@ -26,6 +26,7 @@
     >
       <slot name="label"></slot>
       <input
+        ref="fileUploader"
```
Comment (jayhof) [inline src/components/VueBucketLoader.vue]: `fileUploader` => `fileInput` would make more sense

File: src/components/VueBucketLoader.vue [hunk 2]
```diff
@@ -82,7 +83,9 @@ export default {
       this.$emit('add-files-before', { files });
 
       files.forEach(async (file) => {
-        if (this.beforeUpload(file)) {
+        if (this.beforeUpload(file)
+          && !this.files.some(currentFile => currentFile.file.name === file)
```
Comment (jayhof) [inline src/components/VueBucketLoader.vue]: You want to add the `file` only if it's not present yet in the `files` array right? How about using `Array.includes()`?

File: src/components/VueBucketLoader.vue [hunk 3]
```diff
@@ -148,6 +151,7 @@ export default {
           this.files.findIndex(item => item === file),
           1,
         );
+        this.$refs.fileUploader.value = '';
```
Comment (jayhof) [inline src/components/VueBucketLoader.vue]: Would it not make more sense to "reset" the file input in `handleFilesDropped()`? Also a comment why we're doing `input.value = ''` might be helpful.

Also check out this discussion: https://stackoverflow.com/questions/20549241/how-to-reset-input-type-file seems like not all browsers support setting the value of a file input. Other suggestions are mentioned in the thread.

File: package.json [hunk 4]
```diff
@@ -49,8 +49,8 @@
     "pre-push": "^0.1.1",
     "vue": "^2.5.13",
     "vue-jest": "^2.2.1",
-    "vue-loader": "15",
-    "vue-template-compiler": "^2.5.13",
+    "vue-loader": "^15.7.0",
```
Comment (I2olanD) [inline package.json]: did you pull latest master? because i already upgraded all packages.

File: yarn.lock [hunk 5]
```diff
@@ -1605,9 +1605,15 @@ camelcase@^5.0.0:
   integrity sha512-IXFsBS2pC+X0j0N/GE7Dm7j3bsEBp+oTpb7F50dwEVX7rf3IgwO9XatnegTsDtniKCUtEJH4fSU6Asw7uoVLfQ==
 
 caniuse-lite@^1.0.30000844:
+<<<<<<< HEAD
```
Comment (I2olanD) [inline yarn.lock]: Please fix merge conflicts. See #28

File: package.json (hunk 4 above)
Comment (I2olanD) [reply package.json]: See #28

File: src/components/VueBucketLoader.vue (hunk 2 above)
Comment (aholzner) [reply src/components/VueBucketLoader.vue]: @jayhof Includes doesn't take a function callback so that we can check if file object has a matching name.

**Comment by aholzner** [comment]
@jayhof I have tested on Safari and Firefox, the last 3 versions, and it seems to be working accordingly.

File: src/components/VueBucketLoader.vue (hunk 2 above)
Comment (jayhof) [reply src/components/VueBucketLoader.vue]: Ah right, I missed that!

File: package.json (hunk 4 above)
Comment (jayhof) [reply package.json]: Looks like you guys were able to resolve this issue, I'll mark it as resolved.

**Comment by jayhof** [comment]
@aholzner LGTM

**Pull request merged**